euphonic/cli/optimise_dipole_parameter.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: UTF-8 -*-
+"""
+Run and time the Ewald sum part of an interpolation calculation for a
+small number of q-points for different values of dipole_parameter (this
+is the balance between the real/reciprocal space cutoffs when
+performing the Ewald sum for polar materials) to determine the optimal
+value
+"""
+
+import argparse
+import time
+import numpy as np
+from euphonic import ForceConstants
+
+
+def main():
+    parser = get_parser()
+    args = parser.parse_args()
+    params = vars(args)
+    params.update({'print_to_terminal': True})
+    calculate_optimum_dipole_parameter(**params)
+
+
+def calculate_optimum_dipole_parameter(
+        filename: str,
+        dipole_parameter_min: float = 0.25,
+        dipole_parameter_max: float = 1.5,
+        dipole_parameter_step: float = 0.25,
+        n: int = 100,
+        print_to_terminal: bool = False):
+    """
+    Calculate the optimum dipole_parameter and other dipole_parameters
+    from the filename castep_bin file
+
+    Parameters
+    ----------
+    filename : str
+        The path and name of the .castep_bin/.check to read from
+    dipole_parameter_min
```

Comment:
Noting that you have removed the default values from the comment her, but also that you have much improved the layout of the def statement making the default implementations clearer. Leaving it up to you as to whether the good pythonista would also like the defaults in the comments.